File: problems/1305_all_elements_in_two_binary_search_trees/Solution.java
import java.util.*;

class Solution {
    public List<Integer> getAllElements(TreeNode root1, TreeNode root2) {
        List<Integer> l1 = new ArrayList<>(), l2 = new ArrayList<>();
        inorder(root1, l1); inorder(root2, l2);
        List<Integer> res = new ArrayList<>();
        int i = 0, j = 0;
        while (i < l1.size() && j < l2.size())
            res.add(l1.get(i) <= l2.get(j) ? l1.get(i++) : l2.get(j++));
        while (i < l1.size()) res.add(l1.get(i++));
        while (j < l2.size()) res.add(l2.get(j++));
        return res;
    }
    void inorder(TreeNode node, List<Integer> list) {
        if (node == null) return;
        inorder(node.left, list); list.add(node.val); inorder(node.right, list);
    }
}

File: problems/1305_all_elements_in_two_binary_search_trees/solution.py
from shared.python.data_structures import TreeNode


class Solution:
    def getAllElements(self, root1: 'TreeNode', root2: 'TreeNode') -> list[int]:
        def inorder(node: 'TreeNode', res: list[int]) -> None:
            if not node:
                return
            inorder(node.left, res)
            res.append(node.val)
            inorder(node.right, res)
        l1: list[int] = []
        l2: list[int] = []
        inorder(root1, l1)
        inorder(root2, l2)
        res: list[int] = []
        i = j = 0
        while i < len(l1) and j < len(l2):
            if l1[i] <= l2[j]:
                res.append(l1[i]); i += 1
            else:
                res.append(l2[j]); j += 1
        res.extend(l1[i:])
        res.extend(l2[j:])
        return res
